cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/OperationalLimitConversion.java

```diff
@@ -22,29 +23,78 @@
  */
 public class OperationalLimitConversion extends AbstractIdentifiedObjectConversion {
 
+    private static final String CURRENT_LIMIT = "CurrentLimit";
+    private static final String LIMIT_TYPE = "limitType";
+    private static final String OPERATIONAL_LIMIT = "Operational limit";
+    private static final String OPERATIONAL_LIMIT_TYPE_NAME = "operationalLimitTypeName";
+    private static final String OPERATIONAL_LIMIT_SUBCLASS = "OperationalLimitSubclass";
+
     public OperationalLimitConversion(PropertyBag l, Context context) {
         super("OperationalLimitSet", l, context);
         // Limit can associated to a Terminal or to an Equipment
         terminalId = l.getId("Terminal");
+        equipmentId = l.getId("Equipment");
+        Terminal terminal = null;
         if (terminalId != null) {
             terminal = context.terminalMapping().find(terminalId);
+            if (terminal != null) {
+                createCurrentLimitsAdder(terminal);
+            }
         }
-        equipmentId = l.getId("Equipment");
-        if (equipmentId != null) {
+        if (terminal == null && equipmentId != null) {
             // The equipment may be a Branch, a Dangling line, a Switch ...
-            branch = context.network().getBranch(equipmentId);
-            if (branch == null) {
-                danglingLine = context.network().getDanglingLine(equipmentId);
-                if (danglingLine == null) {
-                    aswitch = context.network().getSwitch(equipmentId);
-                }
+            Identifiable identifiable = context.network().getIdentifiable(equipmentId);
+            if (identifiable instanceof Branch) {
+                Branch branch = context.network().getBranch(equipmentId);
+                createCurrentLimitsAdder(branch);
+            } else if (identifiable instanceof DanglingLine) {
+                DanglingLine danglingLine = context.network().getDanglingLine(equipmentId);
+                createCurrentLimitsAdder(danglingLine);
+            } else if (identifiable instanceof Switch) {
+                Switch aswitch = context.network().getSwitch(equipmentId);
+                notAssigned(aswitch);
+            } else {
+                notAssigned();
             }
```

Comment:
Maybe you could merge this block of code with the createCurrentLimitAdder method?

Reply:
Even if the checking is alike, the generation of `currentLimitAdder` is different so I don't think it can be merged